app/src/main/java/org/simple/clinic/monthlyreports/form/compose/QuestionnaireInputField.kt

```diff
@@ -32,11 +35,10 @@ fun InputField(
         selection = TextRange(initValue.length)
     ))
   }
-  val pattern = remember { Regex("^\\d+\$") }
   TextField(
       value = text,
       onValueChange = {
-        if (it.text.isEmpty() || it.text.matches(pattern)) {
+        if (allowUserInput(inputType = inputFieldComponentData.type, text = it.text)) {
```

Comment:
instead of passing the text, can we infer it using `it.text` inside the private function?
That'll make this line a bit shorter. Also, can't we just call like this?
```
allowUserInput(inputFieldComponentData.type, it.text)
```
Why the need for named args?